src/NSwag.CodeGeneration/Models/ResponseModelBase.cs

```diff
@@ -85,9 +85,16 @@ public bool IsDate
         }
 
         /// <summary>Gets a value indicating whether the response requires a text/plain content.</summary>
-        public bool IsPlainText =>
-            !_response.Content.ContainsKey("application/json") &&
-            (_response.Content.ContainsKey("text/plain") || _operationModel.Produces == "text/plain");
+        public bool IsPlainText
```

Comment:
Is this necessary? 
Changing this is dangerous and a potential breaking change.

Reply:
Yes.
See https://github.com/RicoSuter/NSwag/issues/2596#issuecomment-872618108

Reply:
The problem is that this might be wrong... response type can be a JSON encoded integer which would lead to IsPlainText == true, no?
I think we need to check the produces/content.. why would the current implementation isPlainText evaluate to true for "application/problem+json"?

Reply:
> 
> 
> The problem is that this might be wrong... response type can be a JSON encoded integer which would lead to IsPlainText == true, no?
> I think we need to check the produces/content.. why would the current implementation isPlainText evaluate to true for "application/problem+json"?

I just added another test to illustrate the problem. The API endpoint in the test can produce multiple response types. If the response indicates success (200), then the response is "text/plain"; otherwise the response is "application/problem+json". However, without this change, when the c# client code is generated, non-success responses are still treated as "text/plain".

Reply:
> ... response type can be a JSON encoded integer

I'm not familiar with this response type. Can you create/describe a sample API endpoint method definition in C# for example?

Thank you.

Reply:
I have resolved this issue in my last commit.

Thank you.

Reply:
I wonder why you need to check ActualTypeSchema at all - ie when the response type is text/plain it's always a plain text response, ie also when there is no response schema or it is just any schema, no? If there is a schema with eg properties but the response type is text/plain then i'd say the spec is actually wrong..

Reply:
... or do you want to go with the assumption that it's actually JSON if there is a spec, even if the response type indicates otherwise?

Reply:
Thank you for following up. I just pushed another commit which removes checking ActualTypeSchema. Thank you.

Reply:
Sorry, my question was not really about ActualTypeSchema (we should use that to resolve refs) but checking the .Type, ie changing the property in the first place?

Reply:
I figured out the underlying issue; `_operationModel.Produces` in [IsPlainText](https://github.com/RicoSuter/NSwag/blob/master/src/NSwag.CodeGeneration/Models/ResponseModelBase.cs#L90) shouldn't be used because [Produces](https://github.com/RicoSuter/NSwag/blob/master/src/NSwag.CodeGeneration/Models/OperationModelBase.cs#L265-L267) returns `SuccessResponse?.Produces`. The generated code for a non-success response, in the case where the success response (ex. text/plain) returns a different mime type than a non-success response (ex. application/problem+json), will erroneously assume the response is the same as the success response type (ex. text/plain), and fail when trying to convert the response.

I have pushed a commit with this fix.

Thank you.